build.gradle

```diff
@@ -84,7 +77,56 @@ task('exportVersion', description: 'Write the current build version out to a pro
     }
 }
 
-task('release', description: 'Release the project and publish to all repos.', group: 'release', dependsOn: [exportVersion, tagVersion/*, publishToPackageRepo*/])
+task('publishToPackageRepo', type: HttpTask, group: 'publishing') {
+    description 'Publishes the DEB and RPM packages to the package repository'
+    dependsOn configurations.debBundle, configurations.rpmBundle, buildAll
+    mustRunAfter tagVersion
+
+    config {
+        if (!project.hasProperty('nexus.openrepose.username')) {
+            throw new GradleException('The "nexus.openrepose.username" property is required')
+        }
+        if (!project.hasProperty('nexus.openrepose.password')) {
+            throw new GradleException('The "nexus.openrepose.password" property is required')
+        }
+
+        request.uri = 'https://nexus.openrepose.org'
+        request.auth.basic getProperty('nexus.openrepose.username') as String, getProperty('nexus.openrepose.password') as String
+        request.contentType = ContentTypes.BINARY[0]
+        request.encoder ContentTypes.BINARY[0], Encoders.&binary
+
+        // The requests are configured here (inside of the config closure) so that they are evaluated and
+        // configured during the execution build phase rather than the configuration build phase.
+        // This allows us to dynamically create requests to upload files that are not determined until after the
+        // configuration phase for certain projects without having to dictate evaluation order between the projects.
```

Comment:
This change solves the issues I mentioned in previous comments. It does kind of rely on the plugin to execute the config closure at an expected time, but... this seems better than the alternative.